### PR TITLE
refactor: use new permissions mechanism v4

### PR DESCRIPTION
**Pull request by Thenkei**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by Thenkei** [comment]
This PR must be merged in the **v9** branch

Is a breaking change, we don't support old permissions (No role ACL) format.

Current state:
- Missing tests on extended `@forestadmin/forestadmin-client` classes
- Missing tests on Authorisation service

I guess **v9** will be released as part of the Conditional Action Approval release.

File: test/forest-admin-client/rendering-permissions.unit.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      };
+
+      const renderingPermissionService = new RenderingPermissionServiceForForestExpress(
+        options,
+        userPermissions,
+      );
+
+      const invalidateCacheStub = jest.spyOn(renderingPermissionService, 'invalidateCache');
+      const loadPermissionsStub = jest.spyOn(renderingPermissionService, 'loadPermissions');
+      loadPermissionsStub.mockResolvedValue(renderingPermissions);
```
Comment (Thenkei) [inline test/forest-admin-client/rendering-permissions.unit.test.js:143]: Mock internal private function. I use a shortcut here. Theoretically with should mock `ForestHttpApi.getRenderingPermissions`, but that is weird to mock such an internal function out of the scope.

**Review by Thenkei — COMMENTED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; The PR tackle one subject only
-  ✅&nbsp; nbsp;Automatic tests
   - ✅&nbsp; Unit tests
   - ✅&nbsp; Store tests
   - ✅&nbsp; Route tests
- ✅&nbsp; Manual tests - I've tested every aspect of permissions
   - ✅&nbsp; Routes: list, details, add, update, delete
   - ✅&nbsp; Actions: trigger/trigger with approval/approve (a few days ago maybe need to be redone)
   - ✅&nbsp; Charts: all types simple + SQL one
   - ✅&nbsp; Segments: Simple + SQL (+ use cases to detect available smart actions when restricted to SQL segments / restricted to multiple segments a.k.a special UNION)
   - ✅&nbsp; Scopes: 🔐  
- 🔴 &nbsp; &nbsp;PR title - We need to change it. BREAKING CHANGE (requires role ACL on the project from this version)

File: src/context/build-services.js [hunk 2]
```diff
@@ -1,29 +1,34 @@
 /* eslint-disable global-require */
+const createForestAdminClient = require('@forestadmin/forestadmin-client').default;
+
 module.exports = (context) =>
   context
     .addInstance('logger', () => require('../services/logger'))
+    .addUsingFunction('forestAdminClient', ({ env, logger }) => createForestAdminClient({
+      forestServerUrl: env.FOREST_URL,
+      envSecret: env.FOREST_ENV_SECRET,
+      logger: (level, ...args) => logger[level.toLowerCase()](...args),
```
Comment (Thenkei) [inline src/context/build-services.js:10]:
```suggestion
      logger: (level, ...args) => env.DEBUG? logger[level.toLowerCase()](...args) : () => {},
```

We don't want debugs info by default.

File: src/context/build-services.js [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
+    .addUsingClass('scopeManager', () => require('../services/scope-manager')) // Will be handle by ForestAdminClient
     .addUsingClass('modelsManager', () => require('../services/models-manager'))
-    .addUsingClass('tokenService', () => require('../services/token'))
-    .addUsingClass('oidcConfigurationRetrieverService', () => require('../services/oidc-configuration-retriever'))
-    .addUsingClass('oidcClientManagerService', () => require('../services/oidc-client-manager'))
-    .addUsingClass('authenticationService', () => require('../services/authentication'))
+    .addUsingClass('tokenService', () => require('../services/token')) // Could be handle by ForestAdminClient
+    .addUsingClass('oidcConfigurationRetrieverService', () => require('../services/oidc-configuration-retriever')) // Could be handle by ForestAdminClient
+    .addUsingClass('oidcClientManagerService', () => require('../services/oidc-client-manager')) // Could be handle by ForestAdminClient
+    .addUsingClass('authenticationService', () => require('../services/authentication')) // Could be handle by ForestAdminClient
```
Comment (Thenkei) [inline src/context/build-services.js:31]: Should we remove those comments for now?

Comment (ghusse) [reply src/context/build-services.js:31]: Yes, I think we should

**Comment by Thenkei** [comment]
Here we go

**Comment by Thenkei** [comment]
BREAKING CHANGE: drop support of projects that are not using roles

**Pull request merged**